lib/ask/runtime/broker.ex

```diff
@@ -147,10 +149,30 @@ defmodule Ask.Runtime.Broker do
     end
   end
 
+  defp match_condition(responses, bucket) do
+    bucket_vars = Map.keys(bucket.condition)
+
+    matches = Enum.all?( bucket_vars, fn var ->
+                Enum.any?(responses, fn res ->
+                  (res.field_name == var) && (res.value == Map.fetch!(bucket.condition, var))
+                end)
+              end)
+              
+    matches
+  end
+
   defp update_respondent(respondent, :end) do
     respondent
     |> Respondent.changeset(%{state: "completed", session: nil, completed_at: Timex.now, timeout_at: nil})
     |> Repo.update
+
+    responses = Repo.all(from r in Response, where: r.respondent_id == ^respondent.id)
+    matching_bucket = Repo.all(from b in QuotaBucket, where: b.survey_id == ^respondent.survey_id)
+                    |> Enum.find( fn bucket -> match_condition(responses, bucket) end )
+
+    if matching_bucket do
+      Ask.Repo.query!("update quota_buckets set count = count + 1 where id = ?", [matching_bucket.id])
```

Comment:
Use `Repo.update_all` to avoid writing all the SQL query (https://hexdocs.pm/ecto/Ecto.Repo.html#c:update_all/3)